docs(k8s): how to update the CRI of a node pool (#1260)
scaleway/docs-content: containers/kubernetes/how-to/change-container-runtime-interface.mdx

---
meta:
  title: How to change the Container Runtime Interface (CRI) of an existing cluster
  description: This page explains how to change the Container Runtime Interface (CRI) of an existing cluster
content:

>   h1: How to change the Containter Runtime Interface (CRI) of an existing cluster

Reviewer: Typo: container

Author: ```suggestion
  h1: How to change the Container Runtime Interface (CRI) of an existing cluster
```

  paragraph: This page explains how to change the Container Runtime Interface (CRI) of an existing cluster
tags: cluster cni docker crio cri-o
dates:
  validation: 2023-01-31
  posted: 2023-01-31
categories:
  - containers
---

A container runtime is, as one of the lowest layers of Kubernetes, the software that (in addition to other tasks) ensures that containers are started and stopped accordingly. Historically, Kubernetes used Docker as the container runtime.  

> However, with ongoing development, it became clear that it was not entirely suitable. With the rapidly evolving world of container runtimes, it became clear that other solutions might be better suited.  

Reviewer: "it became clear" repeated twice in paragraph

> The Container Runtime Interface (CRI) was developed as a mean to open up Kubernetes to other container runtimes. CRI is a plugin within a Kubernetes cluster that allows the `kubelet` to use more than one type of container runtime. It consists of a protocol buffer and a gRPC API, as well as libraries, with additional specifications and tools.

Reviewer 2: ```suggestion
The Container Runtime Interface (CRI) was developed as a means to open up Kubernetes to other container runtimes. CRI is a plugin within a Kubernetes cluster that allows the `kubelet` to use more than one type of container runtime. It consists of a protocol buffer and a gRPC API, as well as libraries, with additional specifications and tools.
```

> At Scaleway, we focus on Containerd, the industry-leading CRI solution. [Containerd](https://containerd.io/) was initially developed by Docker and donated to the Cloud Native Computing Foundation (CNCF). It uses a subset of the original Docker engine and comes with most of Docker's functionality for running containers and managing storage and images - but lacks many developer-oriented features. It is, therefore, suitable for large-scale use as part of a container orchestrator, such as Kubernetes. 

Reviewer 2: ```suggestion
At Scaleway, we favor Containerd, the industry-leading CRI solution. [Containerd](https://containerd.io/) was initially developed by Docker and donated to the Cloud Native Computing Foundation (CNCF). It uses a subset of the original Docker engine and comes with most of Docker's functionality for running containers and managing storage and images - but lacks many developer-oriented features. It is, therefore, suitable for large-scale use as part of a container orchestrator, such as Kubernetes. 
```

<Message type="important">
  With Kubernetes version 1.25, Scaleway **Kubernetes Kosmos** will deprecate all Kubernetes interfaces named `docker` and `cri-o`.
</Message>

If you are still using one of the deprecated CRIs, your cluster will have the following limitations:
* Pools using deprecated CRIs cannot scale up.
* Control planes cannot be upgraded if the CRI used by a pool is deprecated.

To migrate your node pools to a supported CRI, you have to create a new resource pool supporting `containerd`, move your pods to the new pool and then delete your old pool.
<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com) 
  - You have [created](/containers/kubernetes/how-to/create-cluster) a Kubernetes Kapsule cluster
</Message>

## How to move your pods to a new pool

1. Click **Kubernetes** in the **Containers** section of the [Scaleway Console](https://console.scaleway.com) side menu. The Kubernetes Kapsule dashboard displays.
2. Click <Icon name="more" /> next to the cluster you want to edit and click **More info** to find detailed information about the cluster.

> 3. Click the **Pools** tab to view the node pools associated to your cluster. 

Reviewer 2: ```suggestion
3. Click the **Pools** tab to view the node pools associated with your cluster. 
```

---
meta:
  title: How to change the Container Runtime Interface (CRI) of an existing node pool
  description: This page explains how to change the Container Runtime Interface (CRI) of an existing node pool
content:
  h1: How to change the Container Runtime Interface (CRI) of an existing node pool
  paragraph: This page explains how to change the Container Runtime Interface (CRI) of an existing node pool
tags: cluster cni docker crio cri-o
dates:
  validation: 2023-01-31
  posted: 2023-01-31
categories:
  - containers
---

> A container runtime is a software that (in addition to other tasks) ensures that containers are started and stopped accordingly. It is one the lowest layers of Kubernetes. Historically, Kubernetes used Docker as a container runtime.

Reviewer 3: ```suggestion
A container runtime is a software that,  in addition to other tasks, ensures that containers are started and stopped accordingly. It is one the lowest layers of Kubernetes. Historically, Kubernetes used Docker as a container runtime.
```

However, with ongoing development, it became clear that it was not entirely suitable. With the rapidly evolving world of container runtimes, other solutions emerged as better suited.  
The Container Runtime Interface (CRI) was developed as a means to open up Kubernetes to other container runtimes. CRI is a plugin within a Kubernetes cluster that allows the `kubelet` to use more than one type of container runtime. It consists of a protocol buffer and a gRPC API, as well as libraries, with additional specifications and tools.

> At Scaleway, we favor Containerd, the industry-leading CRI solution. [Containerd](https://containerd.io/) was initially developed by Docker and donated to the Cloud Native Computing Foundation (CNCF). It uses a subset of the original Docker engine and comes with most of Docker's functionality for running containers and managing storage and images - but lacks many developer-oriented features. It is, therefore, suitable for large-scale use as part of a container orchestrator, such as Kubernetes. 

Reviewer 3: ```suggestion
At Scaleway, we favor Containerd, the industry-leading CRI solution. [Containerd](https://containerd.io/) was initially developed by Docker and donated to the Cloud Native Computing Foundation (CNCF). It uses a subset of the original Docker engine and comes with most of Docker's functionalities for running containers and managing storage and images - but lacks many developer-oriented features. It is, therefore, suitable for large-scale use as part of a container orchestrator, such as Kubernetes. 
```
